js/src/common/classes/GAClient.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Internal dependencies
+ */
+import { debounce } from '../helpers';
+
+// @ts-ignore
+window.dataLayer = window.dataLayer || [];
+
+/**
+ * Genesis Analytics Client
+ *
+ * Forked from BMO's work in Genesis Blocks.
+ *
+ * Follows the singleton pattern to prevent multiple instances of the GA Client from being used.
+ * https://developers.google.com/analytics/devguides/collection/gtagjs
+ */
+export default class GAClient {
+	/**
+	 * Is Google Analytics enabled.
+	 *
+	 * @type {boolean}
+	 */
+	enabled = false;
+
+	/**
+	 * Google Analytics Client
+	 *
+	 * @type {Object}
+	 */
+	client;
+
+	/**
+	 * Google Analytics Measurment ID.
+	 *
+	 * Todo: update this for GCB.
+	 *
+	 * @type {string}
+	 */
+	GA_ID = 'UA-12345';
+
+	/**
+	 * Class constructor.
+	 */
+	constructor() {
+		this.client = function() {
+			// @ts-ignore
+			window.dataLayer.push( arguments );
+		};
+
+		// @ts-ignore
+		this.config = window.gcbAnalyticsConfig || {};
+		if ( this.config.ga_opt_in ) {
+			this.enableAnalytics( this.config.ga_opt_in );
+			this.initClient();
+		}
+	}
+
+	/**
+	 * Enables Google Analytics.
+	 * Setting this value allows the GA Client to respect any opt out configuration.
+	 *
+	 * https://developers.google.com/analytics/devguides/collection/gtagjs/user-opt-out
+	 *
+	 * @param {boolean | number | string} enable The value to be set.
+	 */
+	enableAnalytics( enable ) {
+		enable = !! +enable;
```

Comment:
I'm not sure I'm familiar with this syntax. What is happening on this line?

Reply:
Ah, I stole [that](https://github.com/studiopress/genesis-blocks/blob/c468ce9c049ad6d0ae260f698f445d5e0c6a9b1a/lib/Analytics/js/gaClient.js#L86) from BMO. 

I didn't know about `+` either.

It's first casting `enable` to a `Number` with `+enable`.

Then, it casts it to a `Boolean` with `!!`.

I think this prevents an `enable` value of `'0'` from being `true`.

Like:

```js
if ( '0' ) {
    console.log( 'This is true' );
}
```

Reply:
Thanks!

Reply:
Of course! I just found out about `+`